distribution/docker/Dockerfile

```diff
@@ -18,15 +18,36 @@
 #
 
 ARG JDK_VERSION=11
-FROM alpine as extractor
 
-ARG VERSION=
+# The platform is explicitly specified as x64 to build the Druid distribution.
+# This is because it's not able to build the distribution on arm64 due to dependency problem of web-console. See: https://github.com/apache/druid/issues/13012
+# Since only java jars are shipped in the final image, it's OK to build the distribution on x64.
+# Once the web-console dependency problem is resolved, we can remove the --platform directive.
+FROM --platform=linux/amd64 maven:3.8.6-jdk-11-slim as builder
 
+# Rebuild from source in this stage
+# This can be unset if the tarball was already built outside of Docker
+ARG BUILD_FROM_SOURCE="true"
+
+RUN export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -qq update \
+    && apt-get -qq -y install --no-install-recommends python3 python3-yaml
+
+COPY . /src
 WORKDIR /src
-COPY ./target/apache-druid-${VERSION}-bin.tar.gz .
+RUN --mount=type=cache,target=/root/.m2 if [ "$BUILD_FROM_SOURCE" = "true" ]; then \
+      mvn -B -ff -q dependency:go-offline \
+      install \
+      -Pdist,bundle-contrib-exts \
+      -Pskip-static-checks,skip-tests \
+      -Dmaven.javadoc.skip=true \
+      ; fi
 
-RUN tar -zxf /src/apache-druid-${VERSION}-bin.tar.gz -C /opt \
- && ln -s /opt/apache-druid-${VERSION} /opt/druid
+RUN --mount=type=cache,target=/root/.m2 VERSION=$(mvn -B -q org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate \
```

Comment:
The path here "/distribution/target/apache-druid-${VERSION}-bin.tar.gz" is set from the first step when we do the build. If BUILD_FROM_SOURCE is false, then this step would fail as well.

Reply:
Version no longer gets set as a build argument. It gets inferred automatically from the version defined in pom.xml

The file now matches the code upstream in 25.0, I did not make any changes to it other than the buildkit version to preserve the change made in #143